### DISCUSSION TITLE
NSGA-II:  one solution variable vs. 10 variables with the same volume

### DISCUSSION THREAD

**Discussion by dinohsu1019** [anyoptimization/pymoo#559] [Q&A]
Hi all, 

With NSGA-II algorithm in the discrete case and thus integer solution variables, if I can use one solution variable with cardinality of 10^10 (10 billions), and alternatively, I can also use 10 variables with cardinalities of 10 each, both with the same volume of 10^10, then which one is better in performance, i.e., better diversity and convergence? 

Please kindly advise, thanks.

**Comment by dinohsu1019** [comment]
I asked ChatGPT 3.5 and Stack overflow AI about this question, it seems:
1. If the all solutions in the search space are feasible, then using multiple variables is better.
2. If number of feasible solutions in the search space is small, then using one variable may be better.
I can give an exmaple about feasible solutions being small: say I need to have weights of granularity of 12th-tiles for 6 subscores to add up to the total score, the count of possible combinations is 6188, the cardinality of the one variable, but if I use 6 variables, the volume of search space is 6^12 = 2985984, which 483 times bigger.

**Comment by blankjul** [comment]
I am not sure what role feasibility should play here. One key difference is large-scale optimization if you are using binary variables.
Using integers and only 10 variables it is a relatively small scaled problem (the search space is both times the same anyway).

**Comment by dinohsu1019** [comment]
> I am not sure what role feasibility should play here. One key difference is large-scale optimization if you are using binary variables. Using integers and only 10 variables it is a relatively small scaled problem (the search space is both times the same anyway).

In this scenario, 10 binary variables has the volume of 1024, which is the same as one variable of 1024 values, but in my combination scenario, infeasibility is huge, only 1/483 of solutions are feasible, the benefit of multiple variables is much less significant, at best.
